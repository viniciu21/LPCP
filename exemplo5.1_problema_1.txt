declaration:
    problem_1: float;
end_declaration

main:
    problem_1 = 10.1 ** 2 - 5.5 + 5; # 101.51
end_main