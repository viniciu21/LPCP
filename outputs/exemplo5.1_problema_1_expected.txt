Declaração de variável: Id "problem_1" (2,5)(True,[(Id "problem_1" (2,5),FloatValue 0.0 (2,16))],[],[],[])
Declaração de variável: Id "a" (3,5)(True,[(Id "problem_1" (2,5),FloatValue 0.0 (2,16)),(Id "a" (3,5),FloatValue 0.0 (3,8))],[],[],[])
Declaração de variável: Id "b" (4,5)(True,[(Id "problem_1" (2,5),FloatValue 0.0 (2,16)),(Id "a" (3,5),FloatValue 0.0 (3,8)),(Id "b" (4,5),IntValue 0 (4,8))],[],[],[])
Declaração de variável: Id "c" (5,5)(True,[(Id "problem_1" (2,5),FloatValue 0.0 (2,16)),(Id "a" (3,5),FloatValue 0.0 (3,8)),(Id "b" (4,5),IntValue 0 (4,8)),(Id "c" (5,5),FloatValue 0.0 (5,8))],[],[],[])
Declaração de variável: Id "d" (6,5)(True,[(Id "problem_1" (2,5),FloatValue 0.0 (2,16)),(Id "a" (3,5),FloatValue 0.0 (3,8)),(Id "b" (4,5),IntValue 0 (4,8)),(Id "c" (5,5),FloatValue 0.0 (5,8)),(Id "d" (6,5),IntValue 0 (6,8))],[],[],[])
Atualização de estado sobre a variável: Id "a" (10,5)(True,[(Id "problem_1" (2,5),FloatValue 0.0 (2,16)),(Id "a" (3,5),FloatValue 10.1 (10,9)),(Id "b" (4,5),IntValue 0 (4,8)),(Id "c" (5,5),FloatValue 0.0 (5,8)),(Id "d" (6,5),IntValue 0 (6,8))],[],[],[])
Atualização de estado sobre a variável: Id "b" (11,5)(True,[(Id "problem_1" (2,5),FloatValue 0.0 (2,16)),(Id "a" (3,5),FloatValue 10.1 (10,9)),(Id "b" (4,5),IntValue 2 (11,9)),(Id "c" (5,5),FloatValue 0.0 (5,8)),(Id "d" (6,5),IntValue 0 (6,8))],[],[],[])
Atualização de estado sobre a variável: Id "c" (12,5)(True,[(Id "problem_1" (2,5),FloatValue 0.0 (2,16)),(Id "a" (3,5),FloatValue 10.1 (10,9)),(Id "b" (4,5),IntValue 2 (11,9)),(Id "c" (5,5),FloatValue 5.5 (12,9)),(Id "d" (6,5),IntValue 0 (6,8))],[],[],[])
Atualização de estado sobre a variável: Id "d" (13,5)(True,[(Id "problem_1" (2,5),FloatValue 0.0 (2,16)),(Id "a" (3,5),FloatValue 10.1 (10,9)),(Id "b" (4,5),IntValue 2 (11,9)),(Id "c" (5,5),FloatValue 5.5 (12,9)),(Id "d" (6,5),IntValue 5 (13,9))],[],[],[])
Atualização de estado sobre a variável: Id "problem_1" (14,5)(True,[(Id "problem_1" (2,5),FloatValue 101.51001 (10,9)),(Id "a" (3,5),FloatValue 10.1 (10,9)),(Id "b" (4,5),IntValue 2 (11,9)),(Id "c" (5,5),FloatValue 5.5 (12,9)),(Id "d" (6,5),IntValue 5 (13,9))],[],[],[])
101.51001
Tokens do programa: [Declaration (1,1),Colon (1,12),Id "problem_1" (2,5),Colon (2,14),Type "float" (2,16),SemiColon (2,21),Id "a" (3,5),Colon (3,6),Type "float" (3,8),SemiColon (3,13),Id "b" (4,5),Colon (4,6),Type "int" (4,8),SemiColon (4,11),Id "c" (5,5),Colon (5,6),Type "float" (5,8),SemiColon (5,13),Id "d" (6,5),Colon (6,6),Type "int" (6,8),SemiColon (6,11),EndDeclaration (7,1),Main (9,1),Colon (9,5),Id "a" (10,5),Assign (10,7),FloatValue 10.1 (10,9),SemiColon (10,13),Id "b" (11,5),Assign (11,7),IntValue 2 (11,9),SemiColon (11,10),Id "c" (12,5),Assign (12,7),FloatValue 5.5 (12,9),SemiColon (12,12),Id "d" (13,5),Assign (13,7),IntValue 5 (13,9),SemiColon (13,10),Id "problem_1" (14,5),Assign (14,15),FloatValue 101.51001 (10,9),SemiColon (14,31),Print "print" (15,5),LeftParenthesis (15,10),FloatValue 101.51001 (10,9),RightParenthesis (15,20),SemiColon (15,21),EndMain (16,1)]